src/main/java/net/minecraftforge/gui/IGuiProvider.java

```diff
@@ -0,0 +1,62 @@
+package net.minecraftforge.gui;
+
+import io.netty.buffer.ByteBuf;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a gui provider- something that activates a gui for a player to use.
+ */
+public interface IGuiProvider<T>
+{
+    /**
+     * Deserializes a buffer from the server into the provider instance.
+     * In practice this is run after an open request to set the owner up on the client.
+     *
+     * @param buffer
+     */
+    IGuiProvider deserialize(ByteBuf buffer);
+
+    /**
+     * Serializes data on the server to send along with the open gui packet.
+     * Use this to add any information you need to open the gui properly on the client.
+     *
+     * @param buffer
+     */
+    void serialize(ByteBuf buffer, Object... extras);
+
+    /**
+     * Gets the gui provider object from the instance, after being set with deserialize.
+     *
+     * @param world The world.
+     * @param player The player instance.
+     * @return An object that "owns" the gui- a tile, an entity, an item.
+     */
+    T getOwner(World world, EntityPlayer player);
```

Comment:
What is this "owner" used for? Why is this a public method? As far as I can see forge never uses it anywhere (and it can't do so meaningfully since it could be any object...).